pkg/service/roommanager.go

```diff
@@ -286,6 +286,8 @@ func (r *RoomManager) StartSession(ctx context.Context, roomName string, pi rout
 			}
 		}
 		r.telemetry.ParticipantLeft(ctx, room.Room, p.ToProto())
+
+		room.RemoveDisallowedSubscriptions(p, disallowedSubscriptions)
```

Comment:
Chose this option of adding an argument to `OnClose`. The other option could have been `room.RemoveDisallowedSubscriptions` calling a method on `Participant` and pass in a participantSid resolver to handle disallowed subscriptions. Two things there though
1. A new method need on `Participant` interface.
2. Object life cycle semantics. Calling a method on `Participant` to do some clean up after `Close` is odd.